spec/dds_md5_reporter_spec.rb

```diff
@@ -6,27 +6,59 @@
   include ActiveSupport::Testing::TimeHelpers
 
   let(:upload_id) { 'UPLOAD_ID' }
+  let(:file_version_id) { 'FILE_VERSION_ID' }
+  let(:file_version) {
+    {
+      "id" => file_version_id,
+      "upload" => {
+        "id" => upload_id
+      }
+    }
+  }
   let(:user_key) { 'USER_KEY' }
   let(:agent_key) { 'AGENT_KEY' }
   let(:dds_api_url) { 'DDS_API_URL' }
   let(:expected_json_headers) {
     { 'Content-Type' => "application/json", 'Accept' => "application/json" }
   }
 
-  describe 'required arguments' do
-    context 'missing upload_id' do
+  let(:expected_api_token) { "abc123xyz" }
+  let(:expected_auth_header) {
+    {
+      'Authorization' => expected_api_token
+    }.merge(expected_json_headers)
+  }
+
+  describe 'instantiation' do
+    context 'with missing keywords' do
       subject { DdsMd5Reporter.new }
       it 'should raise an error' do
         expect {
           subject
-        }.to raise_error ArgumentError, "missing keywords: upload_id, user_key, agent_key, dds_api_url"
+        }.to raise_error ArgumentError, "missing keywords: file_version_id, user_key, agent_key, dds_api_url"
+      end
+    end
+
+    context 'with nil file_version_id' do
+      subject {
+        DdsMd5Reporter.new(
+          file_version_id: nil,
+          user_key: user_key,
+          agent_key: agent_key,
+          dds_api_url: dds_api_url
+        )
+      }
+      it 'should raise an error' do
+        expect {
+          subject
+        }.to raise_error ArgumentError, "missing file_version_id, file_version_id, user_key, agent_key, and dds_api_url cannot be nil"
```

Comment:
I find this error message confusing to read, mostly because of the first comma. Maybe something like `missing file_version_id. The keywords file_version_id, user_key, agent_key, and dds_api_url cannot be nil.`

Reply:
ok